src/main/java/org/embulk/input/gcs/SingleFileProvider.java

```diff
@@ -50,94 +48,63 @@ public InputStream openNext() throws IOException
+        return new BufferedInputStream(new FileInputStream(tempFile));
     }
 
     @Override
     public void close()
     {
     }
 
-    @VisibleForTesting
-    static class GcsInputStreamReopener
-            implements ResumableInputStream.Reopener
+    private Void getRemoteContentsWithRetry(final File tempFile, final Storage client, final String bucket, final String key, int maxConnectionRetry)
     {
-        private final Logger log = Exec.getLogger(GcsInputStreamReopener.class);
-        private final File tempFile;
-        private final Storage client;
-        private final String bucket;
-        private final String key;
-        private final int maxConnectionRetry;
-
-        public GcsInputStreamReopener(File tempFile, Storage client, String bucket, String key, int maxConnectionRetry)
-        {
-            this.tempFile = tempFile;
-            this.client = client;
-            this.bucket = bucket;
-            this.key = key;
-            this.maxConnectionRetry = maxConnectionRetry;
-        }
-
-        @Override
-        public InputStream reopen(final long offset, final Exception closedCause) throws IOException
-        {
-            try {
-                return retryExecutor()
-                        .withRetryLimit(maxConnectionRetry)
-                        .withInitialRetryWait(500)
-                        .withMaxRetryWait(30 * 1000)
-                        .runInterruptible(new RetryExecutor.Retryable<InputStream>() {
-                            @Override
-                            public InputStream call() throws IOException
-                            {
-                                log.warn(String.format("GCS read failed. Retrying GET request with %,d bytes offset", offset), closedCause);
-                                Storage.Objects.Get getObject = client.objects().get(bucket, key);
-
-                                try (BufferedOutputStream outputStream = new BufferedOutputStream(new FileOutputStream(tempFile))) {
-                                    IOUtils.copy(getObject.executeMediaAsInputStream(), outputStream);
-                                }
-                                return new BufferedInputStream(new FileInputStream(tempFile));
+        try {
+            return retryExecutor()
+                    .withRetryLimit(maxConnectionRetry)
+                    .withInitialRetryWait(500)
+                    .withMaxRetryWait(30 * 1000)
+                    .runInterruptible(new RetryExecutor.Retryable<Void>() {
+                        @Override
+                        public Void call() throws IOException
+                        {
+                            Storage.Objects.Get getObject = client.objects().get(bucket, key);
+                            try (BufferedOutputStream outputStream = new BufferedOutputStream(new FileOutputStream(tempFile))) {
+                                IOUtils.copy(getObject.executeMediaAsInputStream(), outputStream);
```

Comment:
When `IOUtils.copy` method is called twice by the retry, we don't need to care of data duplication in the existing same file?

Reply:
I think it's ok since plugin will create new FileOutputStream with override mode every time so that file will be overwritten.

Reply:
Got it. Then, looks good.